### ISSUE TITLE
"//" keys should be ignored in "resolution" section of package.json

### ISSUE THREAD

**Issue by Peeja** [yarnpkg/yarn#4774]
**What is the current behavior?**

`"//"` keys (comments) in the `"resolution"` section are not ignored.

Steps to reproduce:
1. Add a key to the `"resolution"` section of a `package.json` like:

    ```
    {
      "resolution": {
        "//": "This should be a comment."
      }
    }
    ```

2. Run `yarn install`.

3. See warning:

    ```
    warning Resolution field "//" does not end with a valid package name and will be ignored
    ```

**What is the expected behavior?**

The key should be ignored. `"//"` keys are a standard way of commenting in `package.json` files in lieu of actual comments being available in JSON. (The `"resolution"` section in particular is a place where comments are necessary, so you can communicate why you're doing heavy-handed dependency tweaking.)

**Please mention your node.js, yarn and operating system version.**

* Node v8.7.0
* Yarn v1.2.1
* macOS 10.12.6 (16G29)

**Comment by rally25rs** [comment]
This should have been fixed a while ago #2484 

It seems to work for me in Yarn 1.2.1

```
$ cat package.json
{
  "name": "yarn-test",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "left-pad": "^1.1.3"
  },
  "resolution": {
    "//": "This should be a comment."
  }
}

$ yarn install
yarn install v1.2.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
✨  Done in 0.12s.
```

**Comment by Peeja** [comment]
@rally25rs Thanks!
